testing/trino-tests/src/test/java/io/trino/execution/resourcegroups/db/TestQueuesDb.java

```diff
@@ -369,6 +371,31 @@ public void testNonLeafGroup()
         assertEquals(queryRunner.getCoordinator().getDispatchManager().getQueryInfo(invalidResourceGroupQuery).getErrorCode(), INVALID_RESOURCE_GROUP.toErrorCode());
     }
 
+    @Test
+    public void testUpdateSoftMemoryLimit()
+            throws Exception
+    {
+        // trigger resource group creation
+        queryRunner.execute("SELECT COUNT(*), clerk FROM orders GROUP BY clerk");
+        InternalResourceGroupManager<?> manager = queryRunner.getCoordinator().getResourceGroupManager().orElseThrow();
+        DbResourceGroupConfigurationManager dbConfigurationManager = (DbResourceGroupConfigurationManager) manager.getConfigurationManager();
+
+        dao.updateResourceGroup(2, "bi-${USER}", "100%", 3, 2, 2, null, null, null, null, null, 1L, TEST_ENVIRONMENT);
+        dbConfigurationManager.load();
+        ResourceGroupInfo groupInfo = manager.tryGetResourceGroupInfo(new ResourceGroupId(new ResourceGroupId("global"), "bi-user"))
+                .orElseThrow(() -> new IllegalStateException("Resource group not found"));
+        assertEquals(groupInfo.getSoftMemoryLimit().toBytes(), queryRunner.getCoordinator().getClusterMemoryManager().getClusterMemoryBytes());
+
+        dao.updateResourceGroup(2, "bi-${USER}", "123MB", 3, 2, 2, null, null, null, null, null, 1L, TEST_ENVIRONMENT);
+        dbConfigurationManager.load();
+
+        // wait for SqlQueryManager which enforce memory limits per second
+        MILLISECONDS.sleep(2000);
```

Comment:
nit: I would prefer to do active wait with shorter sleeps, to finish test as quickly as possible. But given other tests use long sleeps already you can leave it as is. Test methods can be cleaned up in batch.

Reply:
> I would prefer to do active wait with shorter sleeps, to finish test as quickly as possible.

Did you mean `MILLISECONDS.sleep(1000);` ? If so, I can change that in this pr. While I won't touch other test methods as they should be take care in a separate pr.

Reply:
> Did you mean MILLISECONDS.sleep(1000); ? If so, I can change that in this pr. While I won't touch other test methods as they should be take care in a separate pr.

This has a risk of becoming flaky then, right? I agree with @losipiuk that active loop with shorter sleeps would be more reliable.

Reply:
> This has a risk of becoming flaky then, right? I agree with losipiuk that active loop with shorter sleeps would be more reliable.

Sorry about misunderstanding... did you mean something like
```
for (int i = 0; i < 2000; i++) {
  MILLISECONDS.sleep(1);
}
```

Reply:
You can use `assertEventually`

```
        assertEventually(
                new Duration(2, TimeUnit.SECONDS),
                new Duration(100, TimeUnit.MILLISECONDS),
                () -> {.... code which does assertions};
````